apps/dpnk/forms.py

```diff
@@ -196,9 +200,48 @@ def __init__(self, *args, **kwargs):
 
     class Meta:
         model = models.Company
-        fields = ("name", "ico")
+        fields = ("ico", "name")
         error_messages = {"ico": {"stdnum_format": models.company.ICO_ERROR_MESSAGE}}
 
+    @staticmethod
+    def _ares_company_by_ico(ico: str):
+        ico = ico.strip()
+        nsmap = {
+            "are": "http://wwwinfo.mfcr.cz/ares/xml_doc/schemas/ares/ares_answer/v_1.0.1"
+        }
+        r = requests.get(f"https://wwwinfo.mfcr.cz/cgi-bin/ares/darv_std.cgi?ico={ico}", timeout=3)
+        tree = etree.fromstring(r.content)
+        root = tree.getroottree()
+
+        company = root.find("//are:Obchodni_firma", namespaces=nsmap)
+
+        if company is not None:
+            return company.text
+
+        return None
+
+    def clean(self):
+        cleaned_data = super(RegisterCompanyForm, self).clean()
+
+        form_ico = cleaned_data["ico"]
+        form_company_name = cleaned_data["name"]
+
+        if not (form_ico or form_company_name):
+            raise ValidationError(_("At least one field has to be filled"))
+
+        if form_ico:
+            ares_company_name = self._ares_company_by_ico(form_ico)
+            if ares_company_name:
+                cleaned_data["name"] = ares_company_name
+
+        return cleaned_data
+
+    def clean_name(self):
```

Comment:
Tohle není rozumní. Každa firma musí mít název.

Reply:
Teoreticky by to mělo být tak, že buď zadá IČO a název se doplní s ARESu, nebo zadá název...čili by se žádná organizace bez názvu neměla objevit...